docs/developer/contributing.md

```diff
@@ -82,6 +82,12 @@ The Makefile provides several targets:
 * `test`: run the tests
 * `lint`: run linting checks
 
+### Compiling on Linux
+Compiling the Agent on Linux requires a couple of extra dependencies.
+These are 
+* [systemd headers](https://github.com/grafana/agent/blob/main/cmd/agent/Dockerfile#L8-L9) for Promtail
+* [bcc tools](https://github.com/grafana/agent/blob/main/cmd/agent/Dockerfile#L12-L13) for the eBPF integration
```

Comment:
Is this also required even if you don't use the eBPF integration?

Reply:
Sorry for the delay Marc! This was implemented in b7bf8ecdc734c2b7328ccc35d0bcc0e1554ad533.

The eBPF integration will be built requiring the `bcc/bcc_common.h` only on Linux and when the `noebpf` tag is missing. Conversely, if the platform is not linux or _is_ Linux but the noebpf tag is passed, then the stub will be used.

Reply:
Is it possible to do in on the other way around? Like you need to specify an `ebpf` tag in case you want to compile agent with ebpf integration?

Reply:
We discussed this with Robert [here](https://github.com/grafana/agent/pull/1478#discussion_r826909113) and ultimately preferred the opt-out approach. We're using opt-out in other places throughout the project (eg. `nodocker`, `norace`, `nonetwork` build tags) so it might be worth to continue that convention.